### DISCUSSION TITLE
Docker VMM on macOS makes immich cannot detect new photos from external libraries

### DISCUSSION THREAD

**Discussion by Yuuikic** [immich-app/immich#14408] [General]
Hi all. I found a problem which may be caused by new feature of Docker Desktop on Mac.

In version 4.35 of Docker Desktop on Mac, a new beta option called [Docker VMM](https://docs.docker.com/desktop/features/vmm/) has been introduced and is enabled by default.

However, I found that under Docker VMM, changes made to directories mounted into containers from the host machine may not fully synchronize with the container. Switching the Virtual Machine Manager to the Apple Virtualization Framework can resolve this issue.

Hope this helps.

![image](https://github.com/user-attachments/assets/70dda374-80b4-46a8-82eb-be7eaefb8574)

More details:
The mounted path can be read by the container, but the issue arises when I add new files or folders to it. The container cannot detect these changes, even though it’s the same path. Furthermore, I discovered that only newly created files or folders fail to synchronize, while deletions or renames are synchronized successfully.

**Comment by bo0tzz** [comment]
Sounds like a docker bug
